packages/spatial/src/renderer/components/GroupComponent.tsx

```diff
@@ -103,6 +116,36 @@ export function addObjectToGroup(entity: Entity, object: Object3D) {
   proxifyVector3WithDirty(TransformComponent.position, entity, TransformComponent.dirtyTransforms, obj.position)
   proxifyQuaternionWithDirty(TransformComponent.rotation, entity, TransformComponent.dirtyTransforms, obj.quaternion)
   proxifyVector3WithDirty(TransformComponent.scale, entity, TransformComponent.dirtyTransforms, obj.scale)
+
+  if (!(object as Mesh).material) return
```

Comment:
this needs to be moved elsewhere, as it is references SourceComponent and other engine speciic logic

moreso, we probably want to move some of the material state, or create new and cleaner ones in the spatial package, according to our engine modularity hierarchy. anything pertaining to rendering should be in the spatial package, and anything pertaining to sources and assets should be in the engine package